### PR TITLE
Changed numbers with decimal handling

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary
* Changed way we handle numbers:
  * Using all decimals to do calculations
  * Showing only 5 decimals in the UI
* Added new truncate function to send the token amount with 8 decimals
* Fixed issue with MAX button
* Removed unused "variant" prop from button
* Migrated some components to TS

### Android

https://user-images.githubusercontent.com/25931366/173893269-034bb5cd-1700-486d-9752-78be595ae010.mp4

android 1
android 2
android 3
android 4

### iOS

https://user-images.githubusercontent.com/25931366/173893502-daababb9-c586-4efe-907a-64a63e55209c.MP4

ios1
ios2

### Regex test
Regex test

### Comparison between real value, old truncate function and new truncate function
Conversion test

## Notion Ticket
[Ticket](https://www.notion.so/860acccea5a8443aa9479260b8a453fb?v=4f9fe61232824265ba5adebf9dadf5f2&p=eb6b9f3710434168bd67c35805ca7572)

### CODE REVIEW

File: src/constants/business.js [hunk 1]
```diff
@@ -1,3 +1,13 @@
+/**
+ * Decimals used to do truncate an amount in the app
+ */
+export const NUMBER_MAX_DECIMALS = 8;
+
+/**
+ * Decimals shown in the app by default
+ */
+export const VISIBLE_DECIMALS = 5;
```
Comment (mattgle) [inline src/constants/business.js:9]: is this number ok?

File: src/screens/flows/Send/utils.ts [hunk 2]
```diff
@@ -1,18 +1,24 @@
 import { DEFAULT_FEE, XTC_FEE } from '@/constants/addresses';
 import { USD_PER_TC } from '@/utils/assets';
 
+// DELETE THIS CONST
```
Comment (mattgle) [inline src/screens/flows/Send/utils.ts:4]: is this comment ok?

**Review by mattgle — APPROVED** [review]
Just 2 questions! Looking good btw

File: src/screens/flows/Send/utils.ts (hunk 2 above)
Comment (ignaciosantise) [reply src/screens/flows/Send/utils.ts:4]: 😆 nice reminder to delete that. I'll delete it now

File: src/constants/business.js (hunk 1 above)
Comment (ignaciosantise) [reply src/constants/business.js:9]: yes, talked with Karla and we'll unify criteria with the web extension!

**Pull request merged**